packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/classcomponent/ClassComponentBuilder.kt

```diff
@@ -146,15 +167,41 @@ class ClassComponentBuilder<SharedObjectType : Any>(
    * Creates the read-only property whose getter takes the caller as an argument.
    */
   inline fun <reified T> Property(name: String, crossinline body: (owner: SharedObjectType) -> T): PropertyComponentBuilderWithThis<SharedObjectType> {
-    return PropertyComponentBuilderWithThis<SharedObjectType>(ownerType, name).also {
+    return PropertyComponentBuilderWithThis<SharedObjectType>(ownerType.kType, name).also {
       it.get(body)
       properties[name] = it
     }
   }
 
   override fun Property(name: String): PropertyComponentBuilderWithThis<SharedObjectType> {
-    return PropertyComponentBuilderWithThis<SharedObjectType>(ownerType, name).also {
+    return PropertyComponentBuilderWithThis<SharedObjectType>(ownerType.kType, name).also {
       properties[name] = it
     }
   }
+
+  /**
+   * Creates object's lifecycle listener that is called right after the first event listener is added for given event.
+   */
+  fun OnStartObservingSync(eventName: String, body: (self: SharedObjectType) -> Unit) {
```

Comment:
I'm not a big fan of adding `Sync` suffix. This will be a bit confusing and it looks like it will be available only inside classes, right? Maybe we should make all observers synchronous instead?
And again, I still think like we should move these functions to instance functions of the modules and shared objects.

Reply:
I've moved those methods to `SharedObject` class